### PR TITLE
sGAR ~ 2024.06.07 ~ validaiton for document other, squashes " " and "Other" bug for excel

### PR DESCRIPTION
**Pull request by ame-test**
Validations the following scenarios:
* Document type is other and document description is blank.
* Document type is not other and document description is something.
* Invalid document types.
* Invalid document descriptions.

Main files:
* `uploadgar/validations.js`
* `validator.js`
* `src/common/utils/transformers.js`.
* `src/app/api/uploadgar/post.controller.js`

**What changes does this PR bring?**

**Definition of Done**
The following need to be checked off as they are complete as part of the
PR process. If you are unable to mark off an item as complete, then state
in the comments why this is the case.
- [ ] Have you built the code locally and confirmed its working?
- [ ] Is the build confirmed to be passing on CI?
- [ ] Have you added enough relevant unit tests for your change?
- [ ] Have you added enough relevant E2E tests for your change?
- [ ] Have you updated any relevant documentation as part of this change?
- [ ] Has this change been tested against the Acceptance Criteria?
- [ ] Have you considered how this will be deployed in SIT/Staging/Production?

### CODE REVIEW

**Comment by ame-test** [comment]
Here are images of error messages added in.
Screenshot 2024-06-07 at 16 52 26
Screenshot 2024-06-07 at 16 51 10
Screenshot 2024-06-07 at 16 46 39
Screenshot 2024-06-07 at 16 46 12

File: src/app/api/uploadgar/validations.js [hunk 1]
```diff
@@ -5,6 +5,14 @@ const airportValidation = require('../../../common/utils/airportValidation');
 const validator = require('../../../common/utils/validator');
 const ValidationRule = require('../../../common/models/ValidationRule.class');
 const { MAX_STRING_LENGTH, MAX_REGISTRATION_LENGTH } = require('../../../common/config/index');
+const documenttype = require('../../../common/seeddata/egar_saved_people_travel_document_type.json');
+
+
+const documentTypes = documenttype
+    .map(documentType => documentType.documenttype)
+    .filter(documentType => Boolean(documentType))
```
Comment (london-aysun) [inline src/app/api/uploadgar/validations.js:13]: These lines seems to be a repeat of the ones in utils/validator.js, can they be refactored?

File: src/common/utils/validator.js [hunk 2]
```diff
@@ -8,6 +8,22 @@ const genderValues = require('../seeddata/egar_gender_choice.json');
 const { MAX_STRING_LENGTH, MAX_REGISTRATION_LENGTH, MAX_EMAIL_LENGTH, USER_FIRST_NAME_CHARACTER_COUNT, USER_SURNAME_CHARACTER_COUNT } = require('../config/index');
 const logger = require('../../common/utils/logger')(__filename);
 const { airportCodeList } = require('../../common/utils/autocomplete');
+const documenttype = require('../../common/seeddata/egar_saved_people_travel_document_type.json');
+
+const documentTypes = documenttype
```
Comment (london-aysun) [inline src/common/utils/validator.js:13]: documentTypes seems to be used only inside isValidDocumentTypes, could this be moved inside the function closer to where it's been used?

Comment (ame-test) [reply src/common/utils/validator.js]: This has been done.

File: src/app/api/uploadgar/validations.js (hunk 1 above)
Comment (ame-test) [reply src/app/api/uploadgar/validations.js]: This has been done.

**Pull request merged**